### PR TITLE
PE-557 Change Flippers to Clippers for stables

### PR DESCRIPTION
**Pull request by gbalabasquer**
# Description

Spell Description:

COB Team Name or Author(s):

# Contribution Checklist

- [ ] First commit title starts with COB Team Name and Collateral type (e.g. `___-TOKEN-X`)
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [ ] Every contract variable/method declared as public/external private/internal
- [ ] Verify expiration (`4 days + 2 hours` monthly and `30 days` for the rest)
- [ ] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [ ] Validate all addresses used are in Kovan changelog
- [ ] Deploy spell to kovan `SOLC_FLAGS="--optimize --optimize-runs=1" dapp --use solc:0.6.11 build --extract && dapp create DssSpell --gas=5000000 --gas-price="$(seth --to-wei 3 "gwei")"`
- [ ] Ensure contract is verified on `kovan` etherscan
- [ ] Change test to use kovan spell address and deploy timestamp
- [ ] Keep `Kovan-DssSpell.sol` and `Kovan-DssSpell.t.sol` the same, but make a copy in `archive`
- [ ] `squash and merge` this PR

### CODE REVIEW

**Comment by e18r** [comment]
@gbalabasquer unless I'm misunderstanding something, you have the title backwards!

**Comment by gbalabasquer** [comment]
> @gbalabasquer unless I'm misunderstanding something, you have the title backwards!

Thanks, it's fixed now and I force pushed a fix in the commit as well.

File: src/Kovan-DssSpell.sol [hunk 1]
```diff
@@ -66,8 +66,20 @@ contract DssSpellAction is DssAction {
...
+    address constant MCD_CLIP_TUSD_A            = 0x9D547d599489B3950485cBa119FC37Bba9c15c13;
+    address constant MCD_CLIP_CALC_TUSD_A       = 0x4AE93701287b8C86f17E5a0Cb4D0732b5ae6EFBD;
+    address constant MCD_CLIP_USDT_A            = 0xBDd2d10dAF8D86dA1f02bB7c7C7841bC9A4F62D4;
+    address constant MCD_CLIP_CALC_USDT_A       = 0xa3a5163Fa4d46D799fE4B036349f0289D69A4445;
+    address constant MCD_CLIP_PAXUSD_A          = 0x3939B686a0A7265512D38Ea3fe700812A703BF31;
+    address constant MCD_CLIP_CALC_PAXUSD_A     = 0x784863edC4C28D73192bf56944D8803c0b5E0CbF;
+    address constant MCD_CLIP_GUSD_A            = 0x448eD0ff4e154C1cBefE2c8057906Dd3dA194dA5;
+    address constant MCD_CLIP_CALC_GUSD_A       = 0x4DD8AaB74a710E7a95937ef1b2618ee76F829Ba6;
+    address constant MCD_CLIP_PSM_USDC_A        = 0xC8Ca47D0AE4193b3f7F813E95669cfB15d922D56;
+    address constant MCD_CLIP_CALC_PSM_USDC_A   = 0x22C3286711bD63D04Da2Ea95C4d7B556B9502a70;
```
Comment (e18r) [inline src/Kovan-DssSpell.sol:82]: wards

```haskell
0x09D45087c035DbcD8d6fB5e9d4c5341b9101E626
└─ externally owned account: 0x09D45087c035DbcD8d6fB5e9d4c5341b9101E626


0xF8D26c26Ac481794E4Aebf4F35B10d8E9748086a
└─ externally owned account: 0xF8D26c26Ac481794E4Aebf4F35B10d8E9748086a


0xedFc36f75faafa80e39cd4623def15da6CF2B5C0
└─ externally owned account: 0xedFc36f75faafa80e39cd4623def15da6CF2B5C0


0x275076c9c101AF880BD944991258d564FA31D61B
└─ externally owned account: 0x275076c9c101AF880BD944991258d564FA31D61B


0x9D547d599489B3950485cBa119FC37Bba9c15c13
└─ externally owned account: 0x9D547d599489B3950485cBa119FC37Bba9c15c13


0x4AE93701287b8C86f17E5a0Cb4D0732b5ae6EFBD
└─ externally owned account: 0x4AE93701287b8C86f17E5a0Cb4D0732b5ae6EFBD


0xBDd2d10dAF8D86dA1f02bB7c7C7841bC9A4F62D4
└─ externally owned account: 0xBDd2d10dAF8D86dA1f02bB7c7C7841bC9A4F62D4


0xa3a5163Fa4d46D799fE4B036349f0289D69A4445
└─ externally owned account: 0xa3a5163Fa4d46D799fE4B036349f0289D69A4445


0x3939B686a0A7265512D38Ea3fe700812A703BF31
└─ externally owned account: 0x3939B686a0A7265512D38Ea3fe700812A703BF31


0x784863edC4C28D73192bf56944D8803c0b5E0CbF
└─ externally owned account: 0x784863edC4C28D73192bf56944D8803c0b5E0CbF


0x448eD0ff4e154C1cBefE2c8057906Dd3dA194dA5
└─ externally owned account: 0x448eD0ff4e154C1cBefE2c8057906Dd3dA194dA5


0x4DD8AaB74a710E7a95937ef1b2618ee76F829Ba6
└─ externally owned account: 0x4DD8AaB74a710E7a95937ef1b2618ee76F829Ba6


0xC8Ca47D0AE4193b3f7F813E95669cfB15d922D56
└─ externally owned account: 0xC8Ca47D0AE4193b3f7F813E95669cfB15d922D56


0x22C3286711bD63D04Da2Ea95C4d7B556B9502a70
└─ externally owned account: 0x22C3286711bD63D04Da2Ea95C4d7B556B9502a70

```

Comment (e18r) [reply src/Kovan-DssSpell.sol:82]: oh, wait, I did that on mainnet lol

Comment (e18r) [reply src/Kovan-DssSpell.sol:82]: :heavy_check_mark: wards

```haskell
0x09D45087c035DbcD8d6fB5e9d4c5341b9101E626
└─ ward: MCD_PAUSE_PROXY


0xF8D26c26Ac481794E4Aebf4F35B10d8E9748086a
└─ ward: MCD_PAUSE_PROXY


0xedFc36f75faafa80e39cd4623def15da6CF2B5C0
└─ ward: MCD_PAUSE_PROXY


0x275076c9c101AF880BD944991258d564FA31D61B
└─ ward: MCD_PAUSE_PROXY


0x9D547d599489B3950485cBa119FC37Bba9c15c13
└─ ward: MCD_PAUSE_PROXY


0x4AE93701287b8C86f17E5a0Cb4D0732b5ae6EFBD
└─ ward: MCD_PAUSE_PROXY


0xBDd2d10dAF8D86dA1f02bB7c7C7841bC9A4F62D4
└─ ward: MCD_PAUSE_PROXY


0xa3a5163Fa4d46D799fE4B036349f0289D69A4445
└─ ward: MCD_PAUSE_PROXY


0x3939B686a0A7265512D38Ea3fe700812A703BF31
└─ ward: MCD_PAUSE_PROXY


0x784863edC4C28D73192bf56944D8803c0b5E0CbF
└─ ward: MCD_PAUSE_PROXY


0x448eD0ff4e154C1cBefE2c8057906Dd3dA194dA5
└─ ward: MCD_PAUSE_PROXY


0x4DD8AaB74a710E7a95937ef1b2618ee76F829Ba6
└─ ward: MCD_PAUSE_PROXY


0xC8Ca47D0AE4193b3f7F813E95669cfB15d922D56
└─ ward: MCD_PAUSE_PROXY


0x22C3286711bD63D04Da2Ea95C4d7B556B9502a70
└─ ward: MCD_PAUSE_PROXY

```

Comment (gbalabasquer) [reply src/Kovan-DssSpell.sol:82]: Hmmm not sure I'm following this

Comment (gbalabasquer) [reply src/Kovan-DssSpell.sol:82]: oh ok sorry hadn't refreshed the new comments.

**Comment by e18r** [comment]
```haskell

dapp-test: rpc block: latest
Running 14 tests for src/Kovan-DssSpell.t.sol:DssSpellTest
[PASS] testSpellIsCast_USDC_A_clip() (gas: 7086247)
[PASS] testSpellIsCast_PAXUSD_A_clip() (gas: 7091984)
[PASS] testOnTime() (gas: 6806131)
[PASS] testFailTooEarly() (gas: 3399)
[PASS] testSpellIsCast_PSM_USDC_A_clip() (gas: 6847136)
[PASS] testSpellIsCast_GENERAL() (gas: 13578391)
[PASS] testSpellIsCast_GUSD_A_clip() (gas: 7096086)
[PASS] testFailWrongDay() (gas: 3531)
[PASS] testSpellIsCast_USDC_B_clip() (gas: 7086599)
[PASS] testCastCost() (gas: 6809816)
[PASS] testSpellIsCast_USDT_A_clip() (gas: 7105195)
[PASS] testSpellIsCast_TUSD_A_clip() (gas: 7131278)
[PASS] testNewChainlogValues() (gas: 6844001)
[PASS] testFailTooLate() (gas: 3751)

```

**Comment by e18r** [comment]
Checked:
- [x] wards
- [x] contracts verified on Etherscan
- [x] contracts' code matches
- [x] test pass
- [x] code makes sense

LGTM

**Review by e18r — DISMISSED** [review]

**Review by naszam — COMMENTED** [review]
Checked:

- dss-Interfaces Ok
- Math Units Ok
- Stable Clips (verified, constructor args ok (vat, spotter, dog, ilk), wards ok, matches dss github)
- Stable Calcs (verified, no constructor args, wards ok, matches dss github (abaci))
- Stable Liq 2.0 Params Ok (`hole` set to 0 and `tip` set to 1 for Kovan)
- Stable Clips Disabled (`stopped = 3`)
- Prev Clips Changes Ok (`tip = 1` for Kovan)
- Changelog Ok (v1.9.0)
- DssExecLib v0.0.6
- addresses_kovan.sol Matches
- Spell Actions Look Good
- Tests Look Good
- Test Pass:
```haskell
dapp-test: rpc block: latest
Running 14 tests for src/Kovan-DssSpell.t.sol:DssSpellTest
[PASS] testSpellIsCast_USDC_A_clip() (gas: 7086362)
[PASS] testSpellIsCast_PAXUSD_A_clip() (gas: 7092099)
[PASS] testOnTime() (gas: 6806131)
[PASS] testFailTooEarly() (gas: 3399)
[PASS] testSpellIsCast_PSM_USDC_A_clip() (gas: 6847136)
[PASS] testSpellIsCast_GENERAL() (gas: 13713607)
[PASS] testSpellIsCast_GUSD_A_clip() (gas: 7096201)
[PASS] testFailWrongDay() (gas: 3531)
[PASS] testSpellIsCast_USDC_B_clip() (gas: 7086714)
[PASS] testCastCost() (gas: 6809816)
[PASS] testSpellIsCast_USDT_A_clip() (gas: 7105310)
[PASS] testSpellIsCast_TUSD_A_clip() (gas: 7131393)
[PASS] testNewChainlogValues() (gas: 6844001)
[PASS] testFailTooLate() (gas: 3751)
```
LGTM :green_circle:

**Review by naszam — APPROVED** [review]
Checked:

- Estimate Deploy Gas Script Changes Ok (Gas Used 4,185,764)
- Deployed Spell Verified
- Deployed Spell Code Matches GitHub
- Dss-Exec-Lib v0.0.6
- Archive Ok
- Test PASS (with deployed spell address)

```haskell
dapp-test: rpc block: latest
Running 14 tests for src/Kovan-DssSpell.t.sol:DssSpellTest
[PASS] testSpellIsCast_USDC_A_clip() (gas: 7095402)
[PASS] testSpellIsCast_PAXUSD_A_clip() (gas: 7101139)
[PASS] testOnTime() (gas: 6815631)
[PASS] testFailTooEarly() (gas: 8399)
[PASS] testSpellIsCast_PSM_USDC_A_clip() (gas: 6856636)
[PASS] testSpellIsCast_GENERAL() (gas: 17589848)
[PASS] testSpellIsCast_GUSD_A_clip() (gas: 7105241)
[PASS] testFailWrongDay() (gas: 8531)
[PASS] testSpellIsCast_USDC_B_clip() (gas: 7095754)
[PASS] testCastCost() (gas: 6819316)
[PASS] testSpellIsCast_USDT_A_clip() (gas: 7114350)
[PASS] testSpellIsCast_TUSD_A_clip() (gas: 7140433)
[PASS] testNewChainlogValues() (gas: 6853501)
[PASS] testFailTooLate() (gas: 8751)
```

LGTM :green_circle:

**Pull request merged**